src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -844,6 +844,10 @@ public function setDefinition($id, Definition $definition)
 
         $id = strtolower($id);
 
+        if ($definition->isDeprecated()) {
+            @trigger_error(sprintf('The service %s relies on a deprecated definition. You should avoid using it.', $id), E_USER_DEPRECATED);
+        }
```

Comment:
That is also what I thought, but maybe then port it into the container itself ? Or should it be enough in the builder only ?